### PR TITLE
Add coin announcement to create signed transaction rpc call

### PR DESCRIPTION
**Pull request by jack60612**
You can use this to easily link transactions together.
*  for pool block claim fees

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1185,14 +1185,30 @@ async def create_signed_transaction(self, request, hold_lock=True):
         if "coins" in request and len(request["coins"]) > 0:
             coins = set([Coin.from_json_dict(coin_json) for coin_json in request["coins"]])
 
+        coin_announcements = None
```
Comment (mariano54) [inline chia/rpc/wallet_rpc_api.py:1188]: Can we use a type annotation here?

File: chia/rpc/wallet_rpc_client.py [hunk 2]
```diff
@@ -180,17 +180,34 @@ async def get_farmed_amount(self) -> Dict:
         return await self.fetch("get_farmed_amount", {})
 
     async def create_signed_transaction(
-        self, additions: List[Dict], coins: List[Coin] = None, fee: uint64 = uint64(0)
+        self,
+        additions: List[Dict],
+        coins: List[Coin] = None,
+        fee: uint64 = uint64(0),
+        coin_announcements: bytes32 = None,
```
Comment (mariano54) [inline chia/rpc/wallet_rpc_client.py:187]: This should be List[bytes32] right?

**Review by mariano54 — CHANGES_REQUESTED** [review]

File: chia/rpc/wallet_rpc_api.py (hunk 1 above)
Comment (jack60612) [reply chia/rpc/wallet_rpc_api.py]: Added

File: chia/rpc/wallet_rpc_client.py (hunk 2 above)
Comment (jack60612) [reply chia/rpc/wallet_rpc_client.py]: Fixed

**Comment by mariano54** [comment]
A few more comments:
- Undo changes to files not relevant to change: `.flake8`, `.isort.cfg`.
- Add a test to `test_full_node_rpc.py` to make sure it is working
- Fix lint until all the checks are green

File: chia/rpc/wallet_rpc_api.py [hunk 3]
```diff
@@ -1185,14 +1185,30 @@ async def create_signed_transaction(self, request, hold_lock=True):
         if "coins" in request and len(request["coins"]) > 0:
             coins = set([Coin.from_json_dict(coin_json) for coin_json in request["coins"]])
 
+        coin_announcements: Set[bytes32] = None
```
Comment (mariano54) [inline chia/rpc/wallet_rpc_api.py:1188]: I think you will have to change this to Optional[Set[bytes32]] because None is not a Set. 
Or initialize to set()

**Review by mariano54 — DISMISSED** [review]

**Review by mariano54 — APPROVED** [review]

**Pull request merged**